<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>581</width>
    <height>413</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>TyQt</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../resources/tyqt.qrc">
    <normaloff>:/tyqt</normaloff>:/tyqt</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QListView" name="boardList">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>200</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>200</width>
        <height>16777215</height>
       </size>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="infoTab">
       <attribute name="title">
        <string>&amp;Information</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Model:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="modelText">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Location:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="locationText">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Serial Number:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="serialText">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>Interfaces:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTreeWidget" name="interfaceTree">
          <property name="uniformRowHeights">
           <bool>true</bool>
          </property>
          <attribute name="headerDefaultSectionSize">
           <number>140</number>
          </attribute>
          <attribute name="headerMinimumSectionSize">
           <number>100</number>
          </attribute>
          <attribute name="headerStretchLastSection">
           <bool>true</bool>
          </attribute>
          <column>
           <property name="text">
            <string>Name</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Path</string>
           </property>
          </column>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="monitorTab">
       <attribute name="title">
        <string>&amp;Monitor</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QPlainTextEdit" name="monitorText">
          <property name="minimumSize">
           <size>
            <width>240</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="contextMenuPolicy">
           <enum>Qt::CustomContextMenu</enum>
          </property>
          <property name="undoRedoEnabled">
           <bool>false</bool>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="textInteractionFlags">
           <set>Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="monitorEdit">
          <property name="placeholderText">
           <string>Send data...</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <layout class="QGridLayout" name="gridLayout" columnstretch="0,0">
            <item row="0" column="1">
             <widget class="QComboBox" name="newlineComboBox">
              <property name="currentIndex">
               <number>1</number>
              </property>
              <item>
               <property name="text">
                <string>No line ending</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Newline (LF)</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Carriage return (CR)</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Both (CRLF)</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QCheckBox" name="echo">
              <property name="text">
               <string>Echo</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QCheckBox" name="autoscroll">
              <property name="text">
               <string>Autoscroll</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="uploadTab">
       <attribute name="title">
        <string>&amp;Upload</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0,1,0">
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Firmware:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="firmwarePath"/>
          </item>
          <item>
           <widget class="QPushButton" name="browseButton">
            <property name="text">
             <string>Browse...</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QCheckBox" name="resetAfterUpload">
          <property name="text">
           <string>Reset device after successful upload</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="logTab">
       <attribute name="title">
        <string>&amp;Log</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QPlainTextEdit" name="logText">
          <property name="contextMenuPolicy">
           <enum>Qt::CustomContextMenu</enum>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="maximumBlockCount">
           <number>1000</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>581</width>
     <height>19</height>
    </rect>
   </property>
   <property name="contextMenuPolicy">
    <enum>Qt::PreventContextMenu</enum>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionUpload"/>
    <addaction name="actionUploadNew"/>
    <addaction name="separator"/>
    <addaction name="actionReset"/>
    <addaction name="actionReboot"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menu_Help">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="actionWebsite"/>
    <addaction name="actionReportBug"/>
    <addaction name="separator"/>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>&amp;View</string>
    </property>
    <addaction name="actionMinimalInterface"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_View"/>
   <addaction name="menu_Help"/>
  </widget>
  <widget class="QStatusBar" name="statusbar">
   <property name="palette">
    <palette>
     <active>
      <colorrole role="WindowText">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>169</red>
         <green>6</green>
         <blue>7</blue>
        </color>
       </brush>
      </colorrole>
     </active>
     <inactive>
      <colorrole role="WindowText">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>169</red>
         <green>6</green>
         <blue>7</blue>
        </color>
       </brush>
      </colorrole>
     </inactive>
     <disabled>
      <colorrole role="WindowText">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>120</red>
         <green>120</green>
         <blue>120</blue>
        </color>
       </brush>
      </colorrole>
     </disabled>
    </palette>
   </property>
   <property name="sizeGripEnabled">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="contextMenuPolicy">
    <enum>Qt::PreventContextMenu</enum>
   </property>
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="floatable">
    <bool>false</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionUpload"/>
   <addaction name="separator"/>
   <addaction name="actionReset"/>
   <addaction name="actionReboot"/>
  </widget>
  <action name="actionUpload">
   <property name="icon">
    <iconset resource="../resources/tyqt.qrc">
     <normaloff>:/upload</normaloff>:/upload</iconset>
   </property>
   <property name="text">
    <string>&amp;Upload</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+U</string>
   </property>
  </action>
  <action name="actionReset">
   <property name="icon">
    <iconset resource="../resources/tyqt.qrc">
     <normaloff>:/reset</normaloff>:/reset</iconset>
   </property>
   <property name="text">
    <string>&amp;Reset</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>&amp;About</string>
   </property>
  </action>
  <action name="actionReboot">
   <property name="icon">
    <iconset resource="../resources/tyqt.qrc">
     <normaloff>:/reboot</normaloff>:/reboot</iconset>
   </property>
   <property name="text">
    <string>&amp;Bootloader</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+B</string>
   </property>
  </action>
  <action name="actionUploadNew">
   <property name="text">
    <string>Upload &amp;New Firmware</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+U</string>
   </property>
  </action>
  <action name="actionWebsite">
   <property name="text">
    <string>&amp;Website</string>
   </property>
  </action>
  <action name="actionReportBug">
   <property name="text">
    <string>&amp;Report a bug</string>
   </property>
  </action>
  <action name="actionMinimalInterface">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Minimal Interface</string>
   </property>
   <property name="toolTip">
    <string>Minimal Interface</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+H</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>boardList</tabstop>
  <tabstop>tabWidget</tabstop>
  <tabstop>modelText</tabstop>
  <tabstop>locationText</tabstop>
  <tabstop>serialText</tabstop>
  <tabstop>interfaceTree</tabstop>
  <tabstop>monitorText</tabstop>
  <tabstop>monitorEdit</tabstop>
  <tabstop>autoscroll</tabstop>
  <tabstop>echo</tabstop>
  <tabstop>newlineComboBox</tabstop>
  <tabstop>firmwarePath</tabstop>
  <tabstop>browseButton</tabstop>
  <tabstop>logText</tabstop>
 </tabstops>
 <resources>
  <include location="../resources/tyqt.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>329</x>
     <y>225</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
